src/sentry/search/events/builder.py

```diff
@@ -1552,6 +1559,138 @@ def get_snql_query(self) -> None:
+                primary = "counter"
+            else:
+                # An orderby that isn't on a function add it to all of them
+                for framework in query_framework.values():
+                    framework.orderby.append(orderby)
+
+        # Pick one arbitrarily, there's no orderby on functions
+        if primary is None:
+            primary = "distribution"
+
+        return primary, query_framework
+
+    def run_query(self, referrer: str, use_cache: bool = False) -> Any:
+        # Need to split orderby between the 3 possible tables
+        # TODO: need to validate orderby, ordering by tag values is impossible unless the values have low cardinality
+        # and we can transform them (eg. project)
+        primary, query_framework = self._create_query_framework()
+
+        value_map = defaultdict(dict)
+        groupby_aliases = [
+            groupby.alias
+            if isinstance(groupby, (AliasedExpression, CurriedFunction))
+            else groupby.name
+            for groupby in self.groupby
+        ]
+        groupby_values = []
+        result = {
+            "data": None,
+            "meta": [],
+        }
+        # We need to run the same logic on all 3 queries, since the `primary` query could come back with no results. The
+        # goal is to get n=limit results from one query, then use those n results to create a condition for the
+        # remaining queries. This is so that we can respect function orderbys from the first query, but also so we don't
+        # get 50 different results from each entity
+        for query_details in [query_framework.pop(primary), *query_framework.values()]:
+            # Only run the query if there's at least one function, can't query without metrics
+            if len(query_details.functions) > 0:
+                select = [
+                    column
+                    for column in self.columns
+                    if not isinstance(column, CurriedFunction) or column in query_details.functions
+                ]
+                if groupby_values:
+                    # We already got the groupby values we want, add them to the conditions to limit our results so we
+                    # can get the aggregates for the same values
+                    where = self.where + [
+                        Condition(
+                            # Tuples are allowed to have multiple types in clickhouse
+                            Function(
+                                "tuple",
+                                [
+                                    groupby.exp
+                                    if isinstance(groupby, AliasedExpression)
+                                    else groupby
+                                    for groupby in self.groupby
+                                ],
+                            ),
+                            Op.IN,
+                            Function("tuple", groupby_values),
```

Comment:
Good point, We probably don't want to allow doing the 10k limit on these table queries. I'll enforce a 51 limit on this builder for now since we only plan to use this for the performance table :+1:

Reply:
Also keep in mind there is a maximum size of the actual query you can build. So if you have 100k conditions, the SQL string will be too large for Clickhouse to process.